sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/model/Table.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.model
+
+import com.squareup.javapoet.ClassName
+import com.squareup.javapoet.MethodSpec
+import com.squareup.javapoet.NameAllocator
+import com.squareup.javapoet.TypeSpec
+import com.squareup.javapoet.TypeVariableName
+import com.squareup.sqldelight.SqliteParser
+import javax.lang.model.element.Modifier
+
+internal class Table(
+    val interfaceClassName: ClassName,
+    val rule: SqliteParser.Create_table_stmtContext,
+    val nameAllocator: NameAllocator
+) {
+  internal val name = rule.table_name().text
+  internal fun column_def() = rule.column_def()
+  internal fun column_def(i: Int) = rule.column_def(i)
+  internal fun sqliteText() = rule.sqliteText()
+
+  internal fun creatorInterface(nameAllocator: NameAllocator): TypeSpec {
```

Comment:
this probably doesn't need to be in this PR but im too focused to learn how to stash only part of this file so yolo